test/cli/testutils.py

```diff
@@ -70,9 +78,31 @@ def __lookup_cppcheck_exe():
     return exe_path
 
 
+def __copy_and_prepare_cppcheck(tmpdir):
+    file_name = "cppcheck"
+    if sys.platform == "win32":
+        file_name += ".exe"
+    exe = __create_temporary_copy(tmpdir, file_name)
```

Comment:
I think an alternative could be:

    shutil.copy(__lookup_cppcheck(), tmpdir)
    exe = os.path.join(tmpdir, 'cppcheck')
    if sys.platform == "win32":
        exe += ".exe"

however I am not sure if the exe should be calculated here also. I suggest the cppcheck() will call __lookup_cppcheck like so:

    exe = __lookup_cppcheck(tmpdir)

and if __lookup_cppcheck() receives a dirname then it will only look for cppcheck binary in that folder.